File: php/features.php
<?php
declare(strict_types=1);

// TemporaryEmailService SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TemporaryEmailServiceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TemporaryEmailServiceBaseFeature();
            case "test":
                return new TemporaryEmailServiceTestFeature();
            default:
                return new TemporaryEmailServiceBaseFeature();
        }
    }
}

File: py/temporaryemailservice_sdk/features.py
# TemporaryEmailService SDK feature factory

from temporaryemailservice_sdk.feature.base_feature import TemporaryEmailServiceBaseFeature
from temporaryemailservice_sdk.feature.test_feature import TemporaryEmailServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TemporaryEmailServiceBaseFeature(),
        "test": lambda: TemporaryEmailServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
